app/models/manageiq/providers/amazon/manager_mixin.rb

```diff
@@ -51,6 +51,23 @@ module ClassMethods
     # Connections
     #
 
+    # Validate Credentials
+    # args:
+    # {
+    #   "region" => "",
+    #   "endpoints" => {
+    #     "ec2" => {
+    #       "access_key" => "",
+    #       "secret_access_key" => "",
+    #     }
+    #   }
+    # }
+    def validate_credentials(args)
+      region = args["region"]
+      ec2_endpoint = args.dig("endpoints", "ec2")
```

Comment:
dig could return nil if a layer is missing, but the next line doesn't check for nil before going deeper.

Reply:
:+1: good call